iothub/service/src/Message.cs

```diff
@@ -1,40 +1,40 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace Microsoft.Azure.Devices
-{
-    using System;
-    using System.Collections.Generic;
-    using System.Diagnostics.CodeAnalysis;
-    using System.IO;
-    using System.Threading;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Threading;
 
-    using Microsoft.Azure.Devices.Common;
-    using Microsoft.Azure.Devices.Common.Exceptions;
-    using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Amqp;
 
+namespace Microsoft.Azure.Devices
+{
     /// <summary>
     /// The data structure represent the message that is used for interacting with IotHub.
     /// </summary>
     public sealed class Message : IDisposable, IReadOnlyIndicator
     {
-        private readonly object messageLock = new object();
-        private volatile Stream bodyStream;
-        private AmqpMessage serializedAmqpMessage;
-        private bool disposed;
-        private bool ownsBodyStream;
-        private int getBodyCalled;
-        private long sizeInBytesCalled;
+        private readonly object _messageLock = new object();
```

Comment:
I know this is existing code, but isn't it better if we use semaphores instead of a random object to maintain locks? 
It also help stay consistent in thread safety practices.

Reply:
sure, I can do that